### ISSUE TITLE
Pytorch: Speed up get_log_probs function

### ISSUE THREAD

**Issue by binhvq** [lopuhin/transformer-lm#5]
Hi lopuhin.
Thanks for your sharing.
I used your model to predict the next word, but I found the forecast speed relatively slow, probably because of the lm.inference.get_log_probs function to predict the probability of all both words in sentences. Meanwhile, the problem of predicting the next word only requires the probability of the last word.
![Screenshot from 2019-05-03 12-48-25](https://user-images.githubusercontent.com/7200425/57121551-8aa96e80-6da2-11e9-8003-a77344aa333d.png)

**Comment by lopuhin** [comment]
>  I found the forecast speed relatively slow

Thanks for feedback. Did you find it slow compared to other similar models / implementations?

> Meanwhile, the problem of predicting the next word only requires the probability of the last word.

Right. But we still need to process all previous words. I see that we could avoid doing softmax for all but the last word, not sure how much difference will it bring.

**Comment by lopuhin** [comment]
FWIW there is a big speedup in text generation here 4c18649391826c2b1590e722b9559fa5f72ced8e - this speeds up generation of multiple tokens, while the single token generation has the same speed.
